### DISCUSSION TITLE
Transactions are getting send but not showing up on Block Explorer and bridged token are not getting received

### DISCUSSION THREAD

**Discussion by FaizanNehal** [zkSync-Community-Hub/zksync-developers#726] [General]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

I created my own zksync ecosystem using the Zk Stack guide. 
When I am bridging my ETH from L1 to L2, the transaction is getting successfully send but I ma not rceeiving the tokens on my Layer-2, also the transaction is not showing up on my block explorer.
The wallet on block explorer is also showing it as empty.

### Additional Details

I created the ecosystem using the command `zk_inception ecosystem create` then initialized it with `zk_inception ecosystem init
` and then finally ran the server using `zk_inception server`
After that I ran the portal with the command `zk_inception portal`
Then I initialized and ran the block explorer using the commands `zk_inception explorer init`, `zk_inception explorer run-backend` and `zk_inception explorer run`.
The screenshots from my terminals are below.

![Screenshot 2024-10-10 at 1 04 43 AM](https://github.com/user-attachments/assets/47d26215-18c4-4e28-acf4-ce3bf86668e9)

Then I bridge my tokens using the portal. I used the rich wallets provided by matter labs.

![Screenshot 2024-10-10 at 12 51 39 AM](https://github.com/user-attachments/assets/d667e108-afa7-4c25-bb6f-b36d18d565f3)

After more than 15 minutes when transaction time is finished, I am checking the transaction in my block explorer and it is showing transaction not found error.

![Screenshot 2024-10-10 at 1 08 55 AM](https://github.com/user-attachments/assets/1eae9a73-7d27-48a5-804b-e8df79930c94)

and funds are not getting arrived in my wallet

![Screenshot 2024-10-10 at 1 11 06 AM](https://github.com/user-attachments/assets/9b5173f5-3e21-47b5-b53f-0b9a0698dc85)

The containers that are running in my docker are as below.

<img width="991" alt="Screenshot 2024-10-10 at 1 07 47 AM" src="https://github.com/user-attachments/assets/5ee937c4-e147-440a-aec1-96ed89efe1ca">

<img width="991" alt="Screenshot 2024-10-10 at 1 07 54 AM" src="https://github.com/user-attachments/assets/6ac30e12-fda9-412c-bf07-8539a868995e">

**Comment by alhenaai** [comment]
It seems like you've followed the correct steps to set up your ZKsync ecosystem and initiate bridging from L1 to L2. However, the issue with transactions not appearing on the block explorer and tokens not being received on L2 could be due to several reasons:

1. **Configuration Issues**: Ensure that all configurations in your `contracts.yaml` and `ZkStack.yaml` files are correctly set, especially the `base_token.address` and `l1.base_token_addr`. These addresses must match the actual contracts deployed on L1 and L2.

2. **Server and Explorer Sync**: Verify that your block explorer service is properly connected to your L2 node. If the explorer is not synced with the L2 node, it won't display transactions. You might need to check the logs for both the explorer backend and the L2 node to identify any connectivity or syncing issues.

3. **Transaction Bridging Logs**: Check the logs for the bridging process. There might be errors or warnings that could give more insight into why the bridging isn't successful. Look for any errors related to token bridging or transaction processing in the logs of the services you've started.

4. **Network Issues**: Since you are on a testnet, network instability or delays could also be a factor. Ensure that your network settings allow for proper communication between all components of your ecosystem.

5. **Docker Container Health**: Ensure all relevant Docker containers are running without issues. A failing container could disrupt the services necessary for successful transaction processing and bridging.

If after checking these areas the issue persists, you might want to restart the ecosystem and ensure all services are initialized in the correct order as per the setup guide. If necessary, redeploy the contracts and reinitialize the services to ensure a clean state.

For further detailed troubleshooting, you might want to consult the logs more closely or reach out for support with specific logs and error messages that you encounter.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by FaizanNehal** [comment]
The number of transactions on my block explorer is also not getting updated.  

![Screenshot 2024-10-10 at 1 13 31 AM](https://github.com/user-attachments/assets/0a43898f-bc08-42d6-bb64-597c02d55c31)


I am also trying to send the transaction from CLI but it is not moving forward from this line. 

![Screenshot 2024-10-10 at 1 21 14 AM](https://github.com/user-attachments/assets/78d89c40-828a-4004-ac36-7b40def844ad)

**Comment by dutterbutter** [comment]
Hey @FaizanNehal  👋  

First and foremost, great post here man! Screenshots are awesome help! 💯 Before debugging with the below steps, if you've spun down your containers I suggest clearing out all old cache and deleting the containers before starting fresh. 

Can you confirm you are able to make request to each of your running chains API? I just want to make sure the chains are up and running as expected. To do so we can first check if the L2 chain is running at port `3050`: 

```curl
curl --request POST \
  --url http://127.0.0.1:3050 \
  --header 'Content-Type: application/json' \
  --data '{
      "jsonrpc": "2.0",
      "id": 1,
      "method": "zks_L1ChainId",
      "params": []
    }'
```

This should respond with the ChainId of your L1. We can also make a similar request to 8545:

```curl
curl --request POST \
  --url http://127.0.0.1:8545 \
  --header 'Content-Type: application/json' \
  --data '{
      "jsonrpc": "2.0",
      "id": 1,
      "method": "eth_chainId",
      "params": []
    }'
```

Again should respond with the chainID. 

If those are correctly responding to requests, we can also peak into the chains logs to see if we any issue when you initiate a transaction. 

To do so you can either use the desktop app (as you have depicted above) by clicking into each container, or in your terminal `docker ps` to get the container ID and then `docker logs -f <container_id_or_name>`. Once you have the docker logs available go ahead and try initiating the tx again and see if you are able to view any error logs that can help us debug. Please post any debug logs you may see.

**Comment by FaizanNehal** [comment]
Hi @dutterbutter 

Upon running the commands these are m output. Both the outputs are kind of same, I think I might have a problem here maybe. 

L2:
<img width="566" alt="image" src="https://github.com/user-attachments/assets/8accf2f7-6d37-4dfc-bb7a-19b0f90996b6">

L1:
<img width="558" alt="image" src="https://github.com/user-attachments/assets/9bbb7545-9596-4f7b-848b-360d94c50a6f">

**Comment by FaizanNehal** [comment]
Hi @dutterbutter 

I think I have figured out what might be the issue here. When I last run my ecosystem the name of chain on my explorer was `faizanchain` and the chain on portal was `testchain`.

I re-ran the ecosystem again from `zk_inception server` command, and now my chain on both explorer and portal are `faizanchain`.

But now I am getting a different internal error while confirming the transaction. Here it is
<img width="1245" alt="image" src="https://github.com/user-attachments/assets/c7a7343b-317b-4eda-9e1d-cfbb3d5f6d13">

I was also getting the same behaviour two days ago, the weird thing is I cannot figure out why I am getting two completely different behaviours very randomly.

**Reply by dutterbutter** [reply]
It is likely due to old cache / artifacts. However we can peek into the logs our chains so we can see what RPC error we are experiencing. 

I would suggest killing the containers (`docker kill <containerID>`) and then deleting all volumes, images, containers and starting fresh. I am unable to reproduce on my side so apologies I dont have a direct solution here.

**Comment by FaizanNehal** [comment]
Hi @dutterbutter 
I cleared the all the volumes, images and containers from my docker and started it over again. I am getting the same RPC error. Upon sending a transaction there is also no error message that is thrown in the logs in my docker. Here is the screenshot of my logs.

<img width="1000" alt="Screenshot 2024-10-11 at 12 48 49 AM" src="https://github.com/user-attachments/assets/5dfae5d6-90ac-4b71-962b-725efd818f32">

**Comment by FaizanNehal** [comment]
So I also tried sending the transaction from the CLI and it is working and showing up in the explorer. The only unusual thing here is, in my output the transaction hash is showing as `0x7f7f1ee38199cdbad51c41c0ac329136cc2145adbbbefa8724b44dc41e9cabc2` while in the explorer it is `0x97e4cb108a0577ff151db08633eba17b148eac6597f51961fc9beb22e74d7650`.
When I am searching the hash `0x7f7f1ee38199cdbad51c41c0ac329136cc2145adbbbefa8724b44dc41e9cabc2` in explorer, it is showing the not found error.

<img width="811" alt="image" src="https://github.com/user-attachments/assets/256d2797-376e-4738-8a2d-b62287965e8d">

![image](https://github.com/user-attachments/assets/bc764e7d-8a7c-45b3-9c18-36e743f82b39)

**Reply by dutterbutter** [reply]
Very interesting sir 🤔 Let me try to reproduce on my end!

**Reply by dutterbutter** [reply]
I have not been able to reproduce on my side. My only suggestion at this time is to spin down all docker and terminal instance of the infrastructure, clear any cache and only follow the steps outlined in this tutorial: https://code.zksync.io/tutorials/custom-zk-chain 

That is, don't spin up an explorer or portal just yet. Get the basic L1 - L2 infrastructure running as expected and then proceed with the following steps described in the README [here](https://github.com/matter-labs/zksync-era/blob/main/zk_toolbox/README.md#portal) and [here](https://github.com/matter-labs/zksync-era/blob/main/zk_toolbox/README.md#explorer).

**Reply by FaizanNehal** [reply]
Hi @dutterbutter 
I am able to run it but the transaction hash in explorer and transaction hash in the command line terminal are different. Is this the expected behaviour or something unusual is happening here. sorry for too many questions, I need to do some automations so therefore need to have this step clear.

![image](https://github.com/user-attachments/assets/9d6e2441-f4ce-46d3-be7d-201e9253c21b)

<img width="1265" alt="image" src="https://github.com/user-attachments/assets/313f6e6e-02f5-49ae-a8e1-57ff410a13c5">


The portal app is still showing the same error

![image](https://github.com/user-attachments/assets/b70a0445-766e-4729-8227-74807b534822)
